### PR TITLE
feat: deploy and undeploy api integration

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/model_version/toggle_deploy_button.tsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        },
+        { toastLifeTimeMs: 60000 }
+      );
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const deployModal = isDeployModalVisible && (
+    <EuiConfirmModal
```
Comment (wanglam) [inline public/components/model_version/toggle_deploy_button.tsx:208]: Could we reuse `ModelVersionDeploymentConfirmModal` here?

File: public/components/model_version/model_version.tsx [hunk 2]
```diff
@@ -29,14 +30,20 @@ import { ModelVersionInformation } from './version_information';
 import { ModelVersionArtifact } from './version_artifact';
 import { ModelVersionTags } from './version_tags';
 import { ModelVersionFormData } from './types';
+import { ModelGroupSearchItem } from '../../apis/model_group';
```
Comment (wanglam) [inline public/components/model_version/model_version.tsx:33]: Is this PR based the latest `feature/model-registry` branch? I think the `apis/model_group` won't exists in that branch.

File: public/components/model_version/toggle_deploy_button.tsx (hunk 1 above)
Comment (ruanyl) [reply public/components/model_version/toggle_deploy_button.tsx:55]: Created a hook `useDeployment` to abstract the deploy/undeploy logic

File: public/apis/model_version.ts [hunk 3]
```diff
@@ -16,9 +16,7 @@ import { InnerHttpProvider } from './inner_http_provider';
 export interface ModelVersionSearchItem {
   id: string;
   name: string;
-  // TODO: the new version details API may not have this field, because model description is on model group level
-  // we should fix this when integrating the new API changes
-  description: string;
+  model_group_id: string;
```
Comment (wanglam) [inline public/apis/model_version.ts:19]: How about use `model_id` here?

File: public/components/model_version/toggle_deploy_button.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+import { MODEL_VERSION_STATE } from '../../../common';
+import { useDeployment } from '../../hooks/use_deployment';
+
+// CREATED,
+// RUNNING,
+// COMPLETED,
+// FAILED,
+// CANCELLED,
+// COMPLETED_WITH_ERROR
```
Comment (wanglam) [inline public/components/model_version/toggle_deploy_button.tsx:17]: Could we remove this part?

File: public/components/model_version/toggle_deploy_button.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      onCancel={() => setIsUndeployModalVisible(false)}
+      onConfirm={onConfirmUndeploy}
+      confirmButtonText="Undeploy"
+      cancelButtonText="Cancel"
+    >
+      <p>This version will be undeployed. You can deploy it again later.</p>
+    </EuiConfirmModal>
+  );
+
+  const toggleButton = useMemo(() => {
```
Comment (wanglam) [inline public/components/model_version/toggle_deploy_button.tsx:86]: How about reuse `isModelDeployable` and `isModelUndeployable` here?

File: public/components/register_model/register_model.tsx [hunk 6]
```diff
@@ -231,6 +231,7 @@ export const RegisterModelForm = ({ defaultValues = DEFAULT_VALUES }: RegisterMo
           // TODO: Fill model format here
           const { config, url } = preTrainedModel;
           form.setValue('modelURL', url);
+          form.setValue('modelFileFormat', 'TORCH_SCRIPT');
```
Comment (wanglam) [inline public/components/register_model/register_model.tsx:234]: How about separate a variable shared in `.getPreTrainedModel$` and `form.setValue`. We can use `variable.toUpperCase()` here

File: test/mocks/model_handlers.ts [hunk 7]
```diff
@@ -58,4 +62,13 @@ export const modelHandlers = [
       })
     );
   }),
+
+  rest.post(`${MODEL_VERSION_LOAD_API_ENDPOINT}/:modelId`, (req, res, ctx) => {
```
Comment (wanglam) [inline test/mocks/model_handlers.ts:66]: Could we add model these handlers to `model_version_handlers` ?

File: public/components/register_model/register_model.tsx (hunk 6 above)
Comment (wanglam) [reply public/components/register_model/register_model.tsx:234]: Just want to refactor code like these:
```tsx
    const preTrainFormat = "torch_script";
    const subscriber = modelRepositoryManager
      .getPreTrainedModel$(nameParams, preTrainFormat)
      .subscribe(
        (preTrainedModel) => {
          // TODO: Fill model format here
          const { config, url } = preTrainedModel;
          form.setValue('modelURL', url);
          form.setValue('modelFileFormat', preTrainFormat.toUpperCase());
```
It's not a critical issue, we can update in the future.

**Review by wanglam — APPROVED** [review]

**Pull request merged**